### PR TITLE
Upgrade for-range loops

### PR DESCRIPTION
**Pull request by kmws**
Hi,

Most of the C++11 for-range loops seem to be doing a copy of object that is currently used. I've made it so it uses a reference and added a const where applicable.

Compiling & tests report no errors.

### CODE REVIEW

**Comment by Minoru** [comment]
Thanks! That's a lot of changes to review—I need to check that you didn't inadvertently change the meaning somewhere. Please bear with me, that might take a few days if I get distracted.

**Comment by kmws** [comment]
All valid points, thanks. Could you take a look at updated version?

File: src/queueloader.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ void queueloader::reload(std::vector<download>& downloads, bool remove_unplayed)
 	std::vector<download> dltemp;
 	std::fstream f;
 
-	for (auto dl : downloads) {
+	for (auto& dl : downloads) {
```
Comment (Minoru) [inline src/queueloader.cpp]: With methods of `download` constified, it seems `auto&` in this file could be changed to `const auto&`.

**Comment by Minoru** [comment]
Yes, that looks about right! You overlooked one file where `auto&` could be changed to `const auto&` now, though. I'll merge this as soon as you fix that.

**Comment by kmws** [comment]
Yes, thanks for double-checking me... Turns out there are some more spots. ;) I hope that's ok this time.

**Comment by Minoru** [comment]
Yep, everything's okay now. Thank you for working on that, @kmws! Hope to see more contributions from you in the future ;) It looks like you might enjoy working on issues tagged "refactoring", for example.

Also, please don't merge master into your "feature branch" if you can avoid it. The merge commit makes the history more confusing and slows down `git bisect`.

**Comment by kmws** [comment]
>Also, please don't merge master into your "feature branch" if you can avoid it. The merge commit makes the history more confusing and slows down git bisect.

Yeah, sorry about that. ;)

**Pull request merged**